### PR TITLE
update repo for current lingui state

### PR DESCRIPTION
**Pull request by thekip**

### CODE REVIEW

File: lingui.config.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+  },
   fallbackLocales: {
     default: 'en'
   },
   catalogs: [
     {
-      path: 'src/translations/locales/{locale}/messages',
-      include: ['src/pages', 'src/components']
+      path: 'src/locales/{locale}',
+      include: ['src/']
```
Comment (thekip) [inline lingui.config.js:22]: just make structure more flat, easier to navigate

File: next.config.js [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
     // your application
     locales: ['en', 'sr', 'es', 'pseudo'],
     defaultLocale: 'en'
-  }
+  },
+  experimental: {
+    swcPlugins: [
+      ['@lingui/swc-plugin', {}],
+    ],
+  },
```
Comment (thekip) [inline next.config.js:13]: The Plugin works only with nextjs13 and above. Although nextjs 12 has SWC support it has some incompatibilities in the API which breaks the plugin.

File: package.json [hunk 3]
```diff
@@ -5,25 +5,25 @@
   "scripts": {
     "dev": "next dev",
     "debug": "NODE_OPTIONS='--inspect' next dev",
-    "build": "yarn lang && next build",
+    "build": "yarn lang:extract && next build",
     "start": "next start",
     "lang:extract": "NODE_ENV=development lingui extract --clean",
-    "lang:compile": "lingui compile",
-    "lang": "yarn lang:extract && yarn lang:compile",
```
Comment (thekip) [inline package.json:12]: Compiling is redundant if you use `@lingui/loader`

File: package.json [hunk 4]
```diff
@@ -5,25 +5,25 @@
...
     "react-dom": "18.2.0"
   },
   "devDependencies": {
-    "@babel/core": "^7.18.5",
-    "@lingui/cli": "^3.13.3",
-    "@lingui/loader": "^3.13.3",
-    "@lingui/macro": "^3.13.3",
+    "@babel/core": "^7.21.0",
+    "@babel/preset-react": "^7.18.6",
+    "@babel/preset-typescript": "^7.21.0",
```
Comment (thekip) [inline package.json:22]: babel dependencies still needed for extractor to work. This already fixed in upcoming lingui v4

File: src/i18n/utils.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  const cardinalRules = new Intl.PluralRules(locale, { type: 'cardinal' });
+  const ordinalRules = new Intl.PluralRules(locale, { type: 'ordinal' });
+
+  i18n.loadLocaleData(locale, {
+    plurals: (n, ordinal) => {
+      return (ordinal ? ordinalRules : cardinalRules).select(n);
+    },
+  });
```
Comment (thekip) [inline src/i18n/utils.ts:22]: We can utilize native `Intl.PluralRules` instead of 3rd party `make-plurals` it has a decent browser support.

**Comment by thekip** [comment]
Also upcoming v4 is bringing special features for nextjs https://github.com/lingui/js-lingui/issues/1458 stay tuned

**Comment by ivandotv** [comment]
build is failing with:
`Module not found: Can't resolve '../locales'`

**Comment by thekip** [comment]
oops, did a refactoring after all and forget to change one import. fixed

File: src/utils.ts [hunk 6]
```diff
@@ -1,23 +1,27 @@
...
-  let data
-  if (isProduction) {
-    data = await import(`./translations/locales/${locale}/messages`)
-  } else {
-    data = await import(
-      `@lingui/loader!./translations/locales/${locale}/messages.po`
-    )
+export function activateLocale(i18n: I18n, locale: string, messages: Messages) {
+  if (locale === 'pseudo') {
+    locale = "en";
```
Comment (thekip) [inline src/utils.ts:12]: it turned out that `Intl` constructors directly support `pseudo` locale, so this substitution is not needed

**Comment by ivandotv** [comment]
@thekip 
Thank you for this!
Repo is still on Nextjs v12.
Is there anything I should pay special attention to when upgrading to V13

Also, please let me know when the new version of lingui is released so I can update the repository.

Next, I'll look into updating the article where it's appropriate.

**Pull request merged**